service/src/main/java/com/zebrunner/reporting/service/TestService.java

```diff
@@ -413,15 +414,28 @@ public boolean existsTestByIdAndTestRunId(long id, long testRunId) {
         return testMapper.getTestsByTestRunId(testRunId);
     }
 
-    public List<TestResult> getTestResultsByTestCaseId(Long testCaseId, Long limit) {
-        List<TestResult> testResults = testMapper.getTestResultsByTestCaseId(testCaseId, limit);
+    public List<TestResult> getLatestTestResultsByTestId(Long testId, Long limit) {
+        Test test = getNotNullTestById(testId);
+        List<TestResult> testResults = testMapper.getTestResultsByStartTimeAndTestCaseId(test.getTestCaseId(), test.getStartTime(), limit);
```

Comment:
Does it make sense to `@Transactional(readOnly = true)` for this method? As far as I can judge, `getNotNullTestById` and `testMapper.getTestResultsByStartTimeAndTestCaseId` methods execute a select query.